ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/filtered_tenants.py

```diff
@@ -169,6 +170,10 @@ def filtered_vrfs(self: SharedUtils, tenant: dict) -> list[dict]:
 
         vrfs: list[dict] = natural_sort(convert_dicts(tenant.get("vrfs", []), "name"), "name")
         for original_vrf in vrfs:
+            if self.wan_role is not None and get(original_vrf, "wan_vni") is None:
```

Comment:
move this towards the end using
`self.wan_role is not None and get(original_vrf, "wan_vni") is not None`

and need to check later

Reply:
@ClausHolbechArista - what we discussed does not work because it means that any VRF, even if filtered using other mechanisms, has a wan_vni, then it is being rendered on the device and Id ont' think this is what we want here. Essentially moving this in the big `or` statement to decide to append makes all WAN routers ending up with all VRFs having a `wan_vni` set, which trigger some other code later and additional config (can demo live)